tutorials/external-metastore-for-dataproc/index.md

```diff
@@ -107,17 +110,67 @@ Create a Cloud Storage bucket for storing test data, run the following command o
     "gs://${PROJECT}-warehouse" 
 
-### Create Cloud SQL instance
 
+### Create private VPC network
+
+Create user defined cloud networking and peer it with Google services
+
+1.  In Cloud Shell, run the following command to create the VPC network:
+
+        gcloud compute networks create ${VPC_NETWORK_NAME} \
+        --subnet-mode=auto \
+        --bgp-routing-mode=global
+
+1.  In Cloud Shell, run the follwing command to reserve the IPs required for the network:
+
+        gcloud compute addresses create \
+        google-managed-services-${VPC_NETWORK_NAME} \
+        --global \
+        --purpose=VPC_PEERING \
+        --prefix-length=16 \
+        --network=projects/${PROJECT}/global/networks/${VPC_NETWORK_NAME}
+
+1.  In Cloud Shell, run the command to setup VPC peering for Google service on the private VPC network:
+
+        gcloud services vpc-peerings connect \
+        --ranges=google-managed-services-${VPC_NETWORK_NAME} \
+        --network=${VPC_NETWORK_NAME} \
+        --project=${PROJECT}
+
+1.  In Cloud Shell, run the following command to enable VM-to-VM communication for Dataproc clusters and CloudSQL.
 
-Create a MySQL on Cloud SQL instance to be used later to host the Hive metastore database.
+        gcloud compute firewall-rules create allow-all-internal \
+        --network projects/${PROJECT}/global/networks/${VPC_NETWORK_NAME} \
+        --allow=all
+
+1.  In Cloud Shell, run the command to enable Private Google access to the two regions that you will use for this tutorial:
+
+        gcloud compute networks subnets update "${VPC_NETWORK_NAME}" \
+        --region=${REGION} \
+        --enable-private-ip-google-access
+        
+        gcloud compute networks subnets update "${VPC_NETWORK_NAME}" \
+        --region=${WORKER_REGION2} \
+        --enable-private-ip-google-access
+
+### Create Cloud SQL instance
+
+Create a [high availability](https://cloud.google.com/sql/docs/mysql/high-availability) MySQL on Cloud SQL instance to be used later to host the Hive metastore database. The CloudSQL instance uses private IPs for added protection.
 
 In Cloud Shell, execute the following command to create a new Cloud SQL instance:
         
-    gcloud sql instances create hive-metastore-db \
+    gcloud beta sql instances create hive-metastore-db \
     --database-version="MYSQL_5_7" \
     --activation-policy=ALWAYS \
-    --zone ${ZONE}
+    --enable-bin-log \
+    --no-assign-ip \
+    --network="projects/${PROJECT}/global/networks/${VPC_NETWORK_NAME}" \
+    --availability-type=regional \
+    --zone="${ZONE}" \
+    --secondary-zone="${ZONE2}"
+
+**Note:** This tutorial uses an empty `root` user password for simplicity. You can add `--root-password=<your-password>` in the command above to set a root user password, and follow the [initialization guide](https://github.com/GoogleCloudDataproc/initialization-actions/tree/master/cloud-sql-proxy#protecting-passwords-with-kms) for CloudSQL proxy on using passwords with Hive.
```

Comment:
"You can add" -->

To follow the best practice for production systems add

Reply:
Done.